File: apps/web/src/api/intelligence.ts
import { useQuery } from '@tanstack/react-query'

const BASE = '/v1'

export interface IntelligenceDocument {
  id: string
  source_id: string
  raw_url: string
  title: string
  body_text: string
  published_at: string | null
  ingested_at: string
  language: string
  source_type: string
  entities_mentioned: string[]
  sentiment_score: number | null
  relevance_score: number | null
}

export interface DocumentSearchResult {
  items: IntelligenceDocument[]
  total: number
  page: number
  limit: number
}

export function useDocuments(params: {
  entity_id?: string
  source_type?: string
  q?: string
  page?: number
  limit?: number
}) {
  const query = new URLSearchParams()
  if (params.entity_id) query.set('entity_id', params.entity_id)
  if (params.source_type) query.set('source_type', params.source_type)
  if (params.q) query.set('q', params.q)
  if (params.page) query.set('page', String(params.page))
  if (params.limit) query.set('limit', String(params.limit))

  return useQuery<DocumentSearchResult>({
    queryKey: ['documents', params],
    queryFn: async () => {
      const res = await fetch(`${BASE}/intelligence/documents?${query}`)
      if (!res.ok) throw new Error('Failed to load documents')
      return res.json()
    },
  })
}

export function useDataSources() {
  return useQuery({
    queryKey: ['data-sources'],
    queryFn: async () => {
      const res = await fetch(`${BASE}/intelligence/sources`)
      if (!res.ok) throw new Error('Failed to load sources')
      return res.json()
    },
  })
}

File: apps/api/schemas/intelligence.py
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel


class DataSourceOut(BaseModel):
    id: UUID
    name: str
    source_type: str
    endpoint_url: str
    fetch_interval_seconds: int
    reliability_score: float
    last_fetched_at: Optional[datetime]
    is_active: bool

    model_config = {"from_attributes": True}


class IntelligenceDocumentOut(BaseModel):
    id: UUID
    source_id: UUID
    raw_url: str
    title: str
    body_text: str
    published_at: Optional[datetime]
    ingested_at: datetime
    language: str
    source_type: str
    entities_mentioned: list
    sentiment_score: Optional[float]
    relevance_score: Optional[float]

    model_config = {"from_attributes": True}


class DocumentSearchResult(BaseModel):
    items: list[IntelligenceDocumentOut]
    total: int
    page: int
    limit: int
